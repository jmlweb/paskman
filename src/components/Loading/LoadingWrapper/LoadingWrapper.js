// @flow
import colors from '../../../styles/colors';
import FullCover from '../../../components/FullCover';

const LoadingWrapper = FullCover.extend`
  position: fixed;
  background: ${colors.whiteA9};
  z-index: 999999;
  overflow: hidden;
`;

export default LoadingWrapper;
