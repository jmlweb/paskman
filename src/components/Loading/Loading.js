import React from 'react';
import LoadingWrapper from './LoadingWrapper/LoadingWrapper';
import LoadingMain from './LoadingMain/LoadingMain';
import LoadingGroups from './LoadingGroups/LoadingGroups';
import { LoadingLeftGroup, LoadingRightGroup } from './LoadingGroup/LoadingGroup';
import { LoadingPrimaryItem, LoadingSecondaryItem } from './LoadingItem/LoadingItem';
import LoadingSVG from './LoadingSVG/LoadingSVG';
import LoadingMsg from './LoadingMsg/LoadingMsg';

const Loading = ({ text }) => (
  <LoadingWrapper>
    <LoadingMain>
      <LoadingGroups>
        <LoadingLeftGroup>
          <LoadingPrimaryItem />
          <LoadingPrimaryItem />
          <LoadingPrimaryItem />
        </LoadingLeftGroup>
        <LoadingRightGroup>
          <LoadingSecondaryItem />
          <LoadingSecondaryItem />
          <LoadingSecondaryItem />
        </LoadingRightGroup>
      </LoadingGroups>
      <LoadingSVG />
    </LoadingMain>
    {text && <LoadingMsg>{text}</LoadingMsg>}
  </LoadingWrapper>
);

export default Loading;
