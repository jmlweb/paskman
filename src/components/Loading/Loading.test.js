import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Loading from './Loading';
import LoadingWrapper from './LoadingWrapper/LoadingWrapper';

configure({ adapter: new Adapter() });

describe('Loading', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Loading />);
  });
  it('Render the component', () => {
    expect(wrapper.find(LoadingWrapper).length).toEqual(1);
  });
  it('Matchs snapshot', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
