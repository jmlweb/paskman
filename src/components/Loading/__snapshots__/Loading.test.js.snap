// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Matchs snapshot 1`] = `
<div
  className="sc-bwzfXH gTtYPo"
>
  <div
    className="sc-htpNat QFXkF"
  >
    <div
      className="sc-bxivhb eekEAM"
    >
      <div
        className="sc-EHOje diSSTp"
      >
        <div
          className="sc-htoDjs dxoUaj"
        />
        <div
          className="sc-htoDjs dxoUaj"
        />
        <div
          className="sc-htoDjs dxoUaj"
        />
      </div>
      <div
        className="sc-bZQynM eNpBeE"
      >
        <div
          className="sc-dnqmqq iXWRzM"
        />
        <div
          className="sc-dnqmqq iXWRzM"
        />
        <div
          className="sc-dnqmqq iXWRzM"
        />
      </div>
    </div>
    <svg
      version="1.1"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <filter
          id="goo"
        >
          <feGaussianBlur
            in="SourceGraphic"
            result="blur"
            stdDeviation="10"
          />
          <feColorMatrix
            in="blur"
            mode="matrix"
            result="goo"
            values="1 0 0 0 0  0 1 0 0 0  0 0 1 0 0  0 0 0 18 -7"
          />
          <feBlend
            in="SourceGraphic"
            in2="goo"
          />
        </filter>
      </defs>
    </svg>
  </div>
</div>
`;
